data/src/main/java/com/microsoft/azure/kusto/data/auth/MsalTokenProviderBase.java

```diff
@@ -41,13 +48,38 @@ private String determineAadAuthorityUrl(String authorityId) {
 
     @Override
     public String acquireAccessToken() throws DataServiceException, DataClientException {
+        initializeCloudInfo();
         IAuthenticationResult accessTokenResult = acquireAccessTokenSilently();
         if (accessTokenResult == null) {
             accessTokenResult = acquireNewAccessToken();
         }
         return accessTokenResult.accessToken();
     }
 
+    protected void initializeCloudInfo() throws DataServiceException, DataClientException {
+        if (cloudInfo != null) {
+            return;
+        }
+        synchronized (this) {
+            if (cloudInfo != null) {
+                return;
+            }

```

Comment:
This should enclose the entire below section

Reply:
I don't understand? 
The rest of the function is within the sync, or did you mean the if? I don't understand

Reply:
Ignore this.
I blame the morning and not enough coffee.